packages/react/src/FieldSet/FieldSet.test.tsx

```diff
@@ -54,4 +54,36 @@ describe('FieldSet', () => {
 
     expect(ref.current).toBe(component)
   })
+
+  it("renders a hint text after the field-set's legend not marked as optional", () => {
+    const { container } = render(<FieldSet legend="Legend" hint="hint text" />)
+
+    const legend = container.querySelector('legend')
+
+    expect(legend).toHaveTextContent('Legend (hint text)')
+  })
+
+  it('renders the default hint text "niet verplicht" after the field-set\'s legend marked as optional', () => {
+    const { container } = render(<FieldSet legend="Legend" optional={true} />)
+
+    const legend = container.querySelector('legend')
+
+    expect(legend).toHaveTextContent('Legend (niet verplicht)')
+  })
+
+  it('renders the provided hint text "optional" after the field-set\'s legend marked as optional', () => {
+    const { container } = render(<FieldSet legend="Legend" optional={true} hint="optional" />)
```

Comment:
Maybe we shouldn't use the string 'optional' here, seeing as the prop is called `optional` as well.